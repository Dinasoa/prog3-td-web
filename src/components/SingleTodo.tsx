import React from "react";

export const SingleTodo = () => {
    return(
        <>
        </>
    )
}