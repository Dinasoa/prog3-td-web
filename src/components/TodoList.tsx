import React from "react"
import { Todo } from "../model";
import '../style/style.css';

interface Props {
    todos: Todo[];
    setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
    changeStatusToDone: (todo:Todo) => void;
    changeStatusTodo: (todo:Todo) => void;
}

export const TodoList: React.FC<Props> = ({ todos, setTodos , changeStatusToDone , changeStatusTodo}) => {
 
    return (
        <>
            <div className="todoList">
                <div className="todos" data-testid="to-do">
                <span className="title" >TO DO</span>
                    {todos.map((todo) =>  todo.isDone == false && (
                        <>
                        <li key={todo.id} data-testid="todo">
                            <input type="checkbox" data-testid = "input-checkbox" onChange={() => changeStatusToDone(todo)}/>
                            {todo.todo}
                        </li>                          
                        </>
                    ))}
                </div>
                <div className="todos">
                <span className="title">DONE</span>
                    {todos.map((todo ,index) => todo.isDone && (
                        <>
                        <li key={todo.id || index} data-testid='done'>
                            <input data-testid='input-done-checkbox' type="checkbox" onChange={() => changeStatusTodo(todo)} checked={true} />
                            {todo.todo}
                        </li>                          
                        </>
                    ))}
                </div>
            </div>
        </>
    )
}