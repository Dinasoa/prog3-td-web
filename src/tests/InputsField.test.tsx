import {fireEvent, render, screen} from "@testing-library/react";
import { InputsField } from "../components/InputsField";
import React, {SetStateAction} from "react";
import renderer from 'react-test-renderer';
import userEvent from "@testing-library/user-event";

it('Input field should match snapshot' , () => {
    const output = renderer.create(<InputsField todo={"todo"} setTodo={() => {}} handleAdd={() => {}}></InputsField>)
    expect(output).toMatchSnapshot() ;
})

it('Entering text ok' , () => {
    let todo: string = 'to do';
    const { getByPlaceholderText } = render(<InputsField todo={todo} setTodo={jest.fn()} handleAdd={() => {}} />);
    const input = getByPlaceholderText('Enter a task');
    userEvent.type(input , 'to do');
    expect(input).toHaveValue('to do')
})

it('Check if the function handleAdd has been called' , () => {
    let item = "todo" ;
    const setTodo = jest.fn();
    const handleAdd = jest.fn() ;
    const app=render(<InputsField todo={item} setTodo={setTodo} handleAdd={handleAdd}/>) ;
    const button = app.getByTestId("submit-button")
    fireEvent.submit(button) ;
    expect(handleAdd).toHaveBeenCalled()
})

it('Entering text in input should update state', () => {
    let todo: string = '';
    const setTodo : React.Dispatch<SetStateAction<string>> = (newTodo: any) => {
        todo = newTodo;
    };
    const { getByPlaceholderText } = render(<InputsField todo={todo} setTodo={setTodo} handleAdd={() => {}} />);
    const input = getByPlaceholderText('Enter a task');
    fireEvent.change(input, { target: { value: 'Test Todo' } });
    expect(todo).toBe('Test Todo');
})

it('Submitting form should call the function handleAdd', () => {
    const handleAdd = jest.fn();
    const { getByTestId } = render(<InputsField todo="" setTodo={() => {}} handleAdd={handleAdd} />);
    const submitButton = getByTestId('submit-button');
    fireEvent.submit(submitButton);
    expect(handleAdd).toHaveBeenCalled();
});

it('Input field and submit button should be displayed', () => {
    const { getByPlaceholderText, getByTestId } = render(<InputsField todo="" setTodo={() => {}} handleAdd={() => {}} />);
    const input = getByPlaceholderText('Enter a task');
    const submitButton = getByTestId('submit-button');
    expect(input).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
});
