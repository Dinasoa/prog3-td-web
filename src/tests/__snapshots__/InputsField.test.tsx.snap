// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field should match snapshot 1`] = `
<form
  className="input"
  onSubmit={[Function]}
>
  <input
    className="input__box"
    onChange={[Function]}
    placeholder="Enter a task"
    type="input"
    value="todo"
  />
  <button
    className="input_submit"
    data-testid="submit-button"
    type="submit"
  >
    Enter
  </button>
</form>
`;
